code/modules/awaymissions/zlevel.dm

```diff
@@ -62,6 +62,13 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
 /proc/seedRuins(list/z_levels = null, budget = 0, whitelist = /area/space, list/potentialRuins = space_ruins_templates)
 	if(!z_levels || !z_levels.len)
 		z_levels = list(1)
+
+	for(var/zl in z_levels)
+		var/turf/T = locate(1, 1, zl)
+		if(!T)
+			WARNING("Z level [zl] does not exist - Not generating ruins")
+			return

```

Comment:
> so that seedRuins doesn't decide to throw out the baby with the bath water because one of the zlevels in a list wasn't valid.

This is actually what I was discussing above with Remie. IMHO `seedRuins` shouldn't run if -any- of the provided z levels are invalid (weather that happens inside `seedRuins` or outside is another thing).